subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -588,12 +588,8 @@ private ResolveState resolveToStateOrLater(final InternalState requestedState) {
                 // Error if we are executing in a user-managed thread.
                 throw new IllegalStateException("The configuration " + identityPath.toString() + " was resolved from a thread not managed by Gradle.");
             } else {
-                // We don't currently have mutable access to the project, so report a deprecation warning and then continue by attempting to acquire mutable access
-                DeprecationLogger.deprecateBehaviour("The configuration " + identityPath.toString() + " was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a different project.")
-                    .willBeRemovedInGradle7()
-                    .withUserManual("viewing_debugging_dependencies", "sub:resolving-unsafe-configuration-resolution-errors")
-                    .nagUser();
-                return owner.getModel().fromMutableState(p -> resolveExclusively(requestedState));
+                // Error if we have no mutable access to the project
+                throw new IllegalStateException("The configuration " + identityPath.toString() + " was resolved without accessing the project in a safe manner.  This may happen when a configuration is resolved from a different project.");
```

Comment:
:x: If we change this, the error message should be changed too because it's sometimes wrong.

See https://discuss.gradle.org/t/resolving-configurations-in-a-single-project-results-in-error/38080

Reply:
I have updated the exception message. Let me know what you think.